src/parser.rs

```diff
@@ -371,19 +378,22 @@ pub struct AVCVideoPacketHeader {
 }
 
 pub fn avc_video_packet_header(input: &[u8]) -> IResult<&[u8], AVCVideoPacketHeader> {
-  let (i, packet_type) = be_u8(input)?;
-  let packet_type = match packet_type {
-    0 => AVCPacketType::SequenceHeader,
-    1 => AVCPacketType::NALU,
-    2 => AVCPacketType::EndOfSequence,
-    _ => return Err(Err::Error(Error::new(input, ErrorKind::Alt))),
-  };
-  let (i, composition_time) = be_i24(i)?;
-  let header = AVCVideoPacketHeader {
-    packet_type,
-    composition_time,
-  };
-  Ok((i, header))
+  be_u8(input).and_then(|(i, packet_type)| {
```

Comment:
Also here maybe a single tuple with the initial byte and the 24 bit composition time?